receiver/hostmetricsreceiver/factory.go

```diff
@@ -94,15 +92,15 @@ func customUnmarshaler(componentViperSection *viper.Viper, intoCfg interface{})
 
 	cfg.Scrapers = map[string]internal.Config{}
 
-	scrapersSection, err := cp.Sub(scrapersKey)
+	scrapersSection, err := componentParser.Sub(scrapersKey)
 	if err != nil {
 		return err
 	}
 	if len(scrapersSection.AllKeys()) == 0 {
 		return errors.New("must specify at least one scraper when using hostmetrics receiver")
 	}
 
-	for key := range componentViperSection.GetStringMap(scrapersKey) {
+	for key := range cast.ToStringMap(componentParser.Get(scrapersKey)) {
```

Comment:
`cast.ToStringMap` returns `map[string]interface{}{}` when failing to do a cast, if `scrapersKey` doesn't have the right type it might lead to confusing errors

Reply:
Nevermind, this seems to be Viper's behavior. If we have something like `Parser.GetStringMap` I think we should return an error on that case though

Reply:
Indeed, but I tried to not add that for the moment.